demisto_sdk/commands/test_content/test_modeling_rule/init_test_data.py

```diff
@@ -0,0 +1,152 @@
+        rich_help_panel='Logging Configuration',
+    ),
+    log_path: Path = typer.Option(
+        None,
+        '-lp', '--log-path',
+        resolve_path=True,
+        show_default=False,
+        help='Path of directory in which you would like to store all levels of logs.',
+        rich_help_panel='Logging Configuration'
+    ),
+    log_file_name: str = typer.Option(
+        'test-modeling-rule.log',
+        '-ln', '--log-name',
+        resolve_path=True,
+        help='The file name (including extension) where log output should be saved to.',
+        rich_help_panel='Logging Configuration'
+    )
+):
+    """
+    Initializes a test data file for a modeling rule
+    """
+    setup_rich_logging(verbosity, quiet, log_path, log_file_name)
+    errors = False
+    mr_content_entities = [ModelingRule(fp.as_posix()) for fp in input]
+    for mr_entity in mr_content_entities:
+        try:
+            all_mr_entity_fields: Set[str] = set()
+            for mr in mr_entity.rules:
+                all_mr_entity_fields = all_mr_entity_fields.union(mr.fields)
+
+            operation_mode = 'create'
+            default_event_mapping = dict.fromkeys(all_mr_entity_fields)
+            default_dataset = mr_entity.rules[0].dataset
+            default_vendor = mr_entity.rules[0].vendor
+            default_product = mr_entity.rules[0].product
+            test_data_file = mr_entity.testdata_path
+            if test_data_file:
+                operation_mode = 'update'
+                logger.info(f'[cyan]Updating test data file: {test_data_file}[/cyan]', extra={'markup': True})
+                test_data = TestData.parse_file(test_data_file)
+                for event_log in test_data.data:
+                    if not event_log.vendor:
+                        event_log.vendor = default_vendor
+                    if not event_log.product:
+                        event_log.product = default_product
+                    if not event_log.dataset:
+                        event_log.dataset = default_dataset
+                # for expected in test_data.expected_values:
+                    new_mapping = default_event_mapping.copy()
+
+                    # remove xdm expected_values fields that are no longer in the rule
+                    if event_log.expected_values:
+                        keys_to_remove = []
+                        for key in event_log.expected_values:
+                            if key not in new_mapping:
+                                keys_to_remove.append(key)
+                        for key in keys_to_remove:
+                            event_log.expected_values.pop(key)
+                        new_mapping.update(event_log.expected_values)
+
+                    event_log.expected_values = new_mapping
+
+                if count > len(test_data.data):
+                    # create the missing templated data and add it to the test data
+                    templated_event_data_to_add = [
```

Comment:
Use itertools.repeat

Reply:
`itertools.repeat` repeats the same object multiple times versus instantiating multiple `EventData` instances which is actually the behavior I want.